v2/kafka-common/src/main/java/com/google/cloud/teleport/v2/kafka/utils/KafkaConfig.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.kafka.utils;
+
+import com.google.cloud.teleport.v2.kafka.options.KafkaReadOptions;
+import com.google.cloud.teleport.v2.kafka.options.KafkaWriteOptions;
+import com.google.cloud.teleport.v2.kafka.values.KafkaAuthenticationMethod;
+import com.google.cloud.teleport.v2.utils.SecretManagerUtils;
+import java.util.HashMap;
+import java.util.Map;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.kafka.common.config.SaslConfigs;
+import org.apache.kafka.common.config.SslConfigs;
+
+/**
+ * The {@link KafkaConfig} is a utility class for constructing properties for Kafka consumers and
+ * producers.
+ */
+public class KafkaConfig {
+  public static Map<String, Object> fromReadOptions(KafkaReadOptions options) {
+    Map<String, Object> properties =
+        from(
+            options.getKafkaReadAuthenticationMode(),
+            options.getSourceKeystoreLocation(),
+            options.getSourceTruststoreLocation(),
+            options.getSourceTruststorePasswordSecretId(),
+            options.getSourceKeystorePasswordSecretId(),
+            options.getSourceKeyPasswordSecretId(),
+            options.getKafkaReadUsernameSecretId(),
+            options.getKafkaReadPasswordSecretId());
+
+    properties.putAll(KafkaCommonUtils.configureKafkaOffsetCommit(options));
+
+    return properties;
+  }
+
+  public static Map<String, Object> fromWriteOptions(KafkaWriteOptions options) {
+    return from(
+        options.getKafkaWriteAuthenticationMethod(),
+        options.getKafkaWriteKeystoreLocation(),
+        options.getKafkaWriteTruststoreLocation(),
+        options.getDestinationTruststorePasswordSecretId(),
+        options.getKafkaWriteKeystorePasswordSecretId(),
+        options.getKafkaWriteKeyPasswordSecretId(),
+        options.getKafkaWriteUsernameSecretId(),
+        options.getKafkaWritePasswordSecretId());
+  }
+
+  private static Map<String, Object> from(
+      String authMode,
+      String keystoreLocation,
+      String truststoreLocation,
+      String truststorePasswordSecretId,
+      String keystorePasswordSecretId,
+      String keyPasswordSecretId,
+      String usernameSecretId,
+      String passwordSecretId) {
+    Map<String, Object> properties = new HashMap<>();
+    if (authMode == null) {
```

Comment:
Need this check here:
```suggestion
    if (authMode == null || authMode.equals(KafkaAuthenticationMethod.NONE)) {
```
Otherwise it'll throw the "Authentication method not supported" exception below for authMode = NONE.

Reply:
Done